packages/nextjs/components/builders/BuildersGrid.tsx

```diff
@@ -0,0 +1,121 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import { normalize } from "viem/ens";
+import { useEnsAvatar, useEnsName } from "wagmi";
+import { useBuildersEvent } from "~~/hooks/useBuildersEvent";
+import { BUILDER_DATA } from "~~/utils/constants";
+
+export function BuildersGrid() {
+  const { events: uniqueEvents, isLoading, error: errorReadingEvents } = useBuildersEvent();
+
+  const getBuilderData = (address: string) => {
+    return BUILDER_DATA[address as keyof typeof BUILDER_DATA];
+  };
+
+  // Component to handle ENS resolution for a single builder
+  const BuilderCard = ({ builderAddress }: { builderAddress: string }) => {
+    const { data: ensName } = useEnsName({
+      address: builderAddress as `0x${string}`,
+      chainId: 1,
+    });
+
+    const { data: ensAvatar } = useEnsAvatar({
+      name: ensName ? normalize(ensName) : undefined,
+      chainId: 1,
+    });
+
+    const builderData = getBuilderData(builderAddress);
+
+    const displayName = ensName || builderData.name;
+
+    return (
+      <div className="card w-86 h-122 shadow-lg hover:shadow-xl transition-shadow duration-300 rounded-xl overflow-hidden relative">
+        <Image
+          src={ensAvatar || "https://avatars.githubusercontent.com/u/111103306?v=4"}
+          alt={displayName}
+          fill
+          className="object-cover"
+          sizes="100%"
+        />
+
+        <div className="absolute bottom-5 left-5 right-5 bg-white rounded-xl p-4">
+          <div className="space-y-2">
+            <h3 className="font-bold text-lg text-black">{displayName}</h3>
+            <p className="text-sm text-purple-600 font-medium">{builderData.title}</p>
+            <p className="text-sm text-gray-600 mb-3 line-clamp-2">{builderData.description}</p>
+
+            <div className="flex items-center justify-between">
+              <div className="flex items-center gap-3">
+                {Object.entries(builderData.social).map(([platform, url]) => (
+                  <SocialIcon key={platform} type={platform} href={url} />
+                ))}
+              </div>
+              <Link
+                href={builderData.link}
+                className="text-xs text-blue-600 hover:text-blue-800 font-medium transition-colors"
+              >
+                View Profile →
+              </Link>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  };
```

Comment:
It's never a good idea to declare components within other components because they will be recreated at every render. Declare this component outside the BuilderGrid scope, or even better, in a different file to improve readability.